elastic/elastic-agent
# [POC] OpenTelemetry Collector mode

## What does this PR do?



## Why is it important?



## Checklist




## Author's Checklist



## How to test this PR locally

Currently requires running as root. You can read a basic log file and see the output in stdout to verify it's working like this.

Here I read my syslog files and add a service.name resource attribute:
``` yaml
receivers:
  filelog:
    include: [ /var/log/system.log ]
    start_at: beginning

processors:
  resource:
    attributes:
    - key: service.name
      action: insert
      value: elastic-otel-test

exporters:
  debug:
    verbosity: detailed
    sampling_initial: 10000
    sampling_thereafter: 10000

service:
  pipelines:
    logs:
      receivers: [filelog]
      processors: [resource]
      exporters:
        - debug
```

```
mage build
sudo ./build/elastic-agent run --config=otel.yml
```


## Related issues


- 

## Use cases



## Screenshots



## Logs



## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...
